src/ice_shelf/MOM_ice_shelf_diag_mediator.F90

```diff
@@ -0,0 +1,758 @@
+!end interface post_data
+
+!> 2D/3D axes type to contain 1D axes handles and pointers to masks
+type, public :: axesType
+  character(len=15) :: id   !< The id string for this particular combination of handles.
+  integer           :: rank !< Number of dimensions in the list of axes.
+  integer, dimension(:), allocatable :: handles !< Handles to 1D axes.
+  type(diag_ctrl), pointer :: diag_cs => null() !< A structure that is used to regulate diagnostic output
+end type axesType
+
+!> This type is used to represent a diagnostic at the diag_mediator level.
+type, private :: diag_type
+  logical :: in_use              !< This diagnostic is in use
+  integer :: fms_diag_id         !< underlying FMS diag id
+  character(len=24) :: name      !< The diagnostic name
+  real :: conversion_factor = 0. !< A factor to multiply data by before posting to FMS, if non-zero.
+  real, pointer, dimension(:,:)   :: mask2d => null()      !< A 2-d mask on the data domain for this diagnostic
+  real, pointer, dimension(:,:)   :: mask2d_comp => null() !< A 2-d mask on the computational domain for this diagnostic
+end type diag_type
+
+!>   The SIS_diag_ctrl data type contains times to regulate diagnostics along with masks and
+!! axes to use with diagnostics, and a list of structures with data about each diagnostic.
+type, public :: diag_ctrl
+  integer :: doc_unit = -1 !< The unit number of a diagnostic documentation file.
+                           !! This file is open if doc_unit is > 0.
+
+  ! The following fields are used for the output of the data.
+  ! These give the computational-domain sizes, and are relative to a start value
+  ! of 1 in memory for the tracer-point arrays.
+  integer :: is  !< The start i-index of cell centers within the computational domain
+  integer :: ie  !< The end i-index of cell centers within the computational domain
+  integer :: js  !< The start j-index of cell centers within the computational domain
+  integer :: je  !< The end j-index of cell centers within the computational domain
+  ! These give the memory-domain sizes, and can be start at any value on each PE.
+  integer :: isd !< The start i-index of cell centers within the data domain
+  integer :: ied !< The end i-index of cell centers within the data domain
+  integer :: jsd !< The start j-index of cell centers within the data domain
+  integer :: jed !< The end j-index of cell centers within the data domain
+  real :: time_int              !< The time interval in s for any fields that are offered for averaging.
+  type(time_type) :: time_end   !< The end time of the valid interval for any offered field.
+  logical :: ave_enabled = .false. !< .true. if averaging is enabled.
+
+  !>@{ The following are 3D and 2D axis groups defined for output.  The names indicate
+  !! the horizontal locations (B, T, Cu, or Cv), vertical locations (L, i, or 1) and
+  !! thickness categories (c, c0, or 1).
+  type(axesType) :: axesBL, axesTL, axesCuL, axesCvL
```

Comment:
None of the axesType variables on lines 79-82 are defined or used with the ice shelves. They should be deleted, at least for now. If we add a 3-d structure to the ice shelves, new types can be added when they are being properly defined.